### PR TITLE
Add minimum popup size

### LINKED ISSUE #135: As a user, I want to see Simple Signer popups with a specific dimension
Background:

The only way to resize popups is when a client implements Simple Signer inside his project.
The objective of this ticket is to give Simple Signer a default size when it's opened.

### PR DESCRIPTION
**Pull request by RivarolaNicolas**
If this PR is accepted the documentation should be changed to explain that when opening the popup through window.open, the developer will have to add the parameters width and height no matter what. If they want to use Simple Signer's default they can leave it like this:

`const connectWindow = window.open(${simpleSignerHost}/connect, 'Connect_Window', 'width, height');`

Else, they will be able to set the desired size this way `const connectWindow = window.open(${simpleSignerHost}/connect, 'Connect_Window', 'width=desiredWidth, height=desiredHeight');`

If no width and height parameters are passed, the popup will open on a new tab or with the latest windows size used by the user.

### CODE REVIEW

**Comment by RivarolaNicolas** [comment]
Minimum size when signing a transaction:

image

Minimum size when connecting:

image

File: src/lib/components/transaction/Transaction.svelte [hunk 1]
```diff
@@ -82,6 +82,17 @@
...
+    const minimumPopupHeight = 570;
+    const minimumPopupWidth = 360;
+
+    if (window.outerHeight < minimumPopupHeight) {
+        self.resizeTo(self.outerWidth, minimumPopupHeight);
+    }
+
+    if (window.outerWidth < minimumPopupWidth) {
+        self.resizeTo(minimumPopupWidth, self.outerHeight);
+    }
```
Comment (fsodano) [inline src/lib/components/transaction/Transaction.svelte]: This does not belong here. The Transaction component has no concept of a `window`. The Sign route would be more appropriate.

File: src/routes/connect/Connect.svelte [hunk 2]
```diff
@@ -27,6 +26,19 @@
...
+    const minimumPopupHeight = 210;
+    const minimumPopupWidth = 340;
+
+    if (window.outerHeight < minimumPopupHeight) {
+        self.resizeTo(self.outerWidth, defaultPopupHeight);
+    }
+
+    if (window.outerWidth < minimumPopupWidth) {
+        self.resizeTo(defaultPopupWidth, self.outerHeight);
+    }
```
Comment (fsodano) [inline src/routes/connect/Connect.svelte]: Please create a helper method that can be called from both routes and handles this logic in a single place

File: src/lib/components/helpers/popUpSizeHelper.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    defaultHeight: number,
+    defaultWidth: number,
+): void {
+    if (window.outerHeight < minimumHeight && window.outerWidth < minimumWidth) {
+        self.resizeTo(defaultWidth, defaultHeight);
+    } else if (window.outerHeight < minimumHeight) {
+        self.resizeTo(window.outerWidth, defaultHeight);
+    } else if (window.outerWidth < minimumWidth) {
+        self.resizeTo(defaultWidth, window.outerHeight);
+    }
```
Comment (fsodano) [inline src/lib/components/helpers/popUpSizeHelper.ts]: This is still not a good way to express this

Consider doing something along these lines:

```
const width = logic to calculate what the final width should be
const height = logic to calculate what the final height should be
self.resizeTo(width, height);
```

**Review by fsodano — APPROVED** [review]

**Review by Genebson — APPROVED** [review]

**Pull request merged**